### ISSUE TITLE
Circom 2.1.0 Anonymous Components and Tuples

### ISSUE THREAD

**Issue by vivianjeng** [Unirep/Unirep#530]
See: https://docs.circom.io/circom-language/anonymous-components-and-tuples/

Now we assigned every input into another templates with
```circom
template A(n){
   signal input a, b;
   signal output c;
   c <== a*b;
}
template B(n){
   signal input in[n];
   signal out;
   component temp_a = A(n);
   temp_a.a <== in[0]; 
   temp_a.b <== in[1];
   out <== temp_a.c;
}
component main = B(2);
```

with anonymous components, we can assign each input with

```circom
template A(n){
   signal input a, b;
   signal output c;
   c <== a*b;
}
template B(n){
   signal input in[n];
   signal out <== A(n)(in[0],in[1]);
}
component main = B(2);
```

which will improve the readability and reduce lines.

**Comment by FoodChain1028** [comment]
Hi @vivianjeng, we are going to work on this issue recently, yet we want to confirm that the version of circom is fixed at 2.1.0 or we can use ^2.1.0 (such as 2.1.1 or higher version)?

**Comment by vivianjeng** [comment]
> Hi @vivianjeng, we are going to work on this issue recently, yet we want to confirm that the version of circom is fixed at 2.1.0 or we can use ^2.1.0 (such as 2.1.1 or higher version)?

I want to change how we use templates.
e.g. now 
```circom
    component epoch_key_hasher = EpochKeyHasher();
    epoch_key_hasher.identity_secret <== identity_secret;
    epoch_key_hasher.attester_id <== attester_id;
    epoch_key_hasher.epoch <== epoch;
    epoch_key_hasher.nonce <== nonce;
```

but we can use the template like this

```circom
(identity_secret, attester_id, epoch, nonce) <== EpochKeyHasher()(identity_secret, attester_id, epoch, nonce);
```

we already use circom version 2.1.5
but it seems like there is a higher version
we can try that too
but maybe in different issue
